com.unity.ml-agents/Runtime/Sensors/AttributeFieldSensor.cs

```diff
@@ -0,0 +1,72 @@
+using System.Reflection;
+
+namespace Unity.MLAgents.Sensors
+{
+    internal class AttributeFieldSensor : ISensor
+    {
+        object m_Object;
+        FieldInfo m_FieldInfo;
+        // Not currently used, but might want later.
+        ObservableAttribute m_ObservableAttribute;
+
+        string m_SensorName;
+        int[] m_Shape;
+
+        public AttributeFieldSensor(object o, FieldInfo fieldInfo, ObservableAttribute observableAttribute)
+        {
+            m_Object = o;
+            m_FieldInfo = fieldInfo;
+            m_ObservableAttribute = observableAttribute;
+
+            m_SensorName = $"ObservableAttribute:{fieldInfo.DeclaringType.Name}.{fieldInfo.Name}";
+            // TODO handle Vector3, quaternion, blittable(?)
```

Comment:
wondering how this will scale for custom data types

Reply:
Code is out of date, but realized I never replied. It's fairly easy to add new types if we see a demand for them. I don't think we should make a framework for letting users register their own type, since there wouldn't be much advantage to doing this vs. writing your own sensor.